lib/Doctrine/Common/Annotations/AnnotationReader.php

```diff
@@ -184,6 +184,14 @@ public function __construct()
     }
 
     /**
+     * @param bool $notImportedAnnotationsIgnored
+     */
+    public function setParserIgnoreNotImportedAnnotations($notImportedAnnotationsIgnored)
```

Comment:
I made changes as you proposed. Injecting parser is the best solution. I wasn't sure if such changes are possible because construction of that object is a bit untidy: instances of parsers are created in constructor - tight coupling.

Thnx for feedback